Undefined name 'system_fonts' in backend_pgf.py

flake8 testing of https://github.com/matplotlib/matplotlib on Python 3.6.3

$ __flake8 . --count --select=E901,E999,F821,F822,F823 --show-source --statistics__
```
./lib/matplotlib/backends/backend_pgf.py:144:20: F821 undefined name 'system_fonts'
    elif family in system_fonts and rcParams["pgf.texsystem"] != "pdflatex":
                   ^
1     F821 undefined name 'system_fonts'
1
```

--- a/lib/matplotlib/backends/backend_pgf.py
+++ b/lib/matplotlib/backends/backend_pgf.py
@@ -141,7 +141,8 @@ def _font_properties_str(prop):
     family = prop.get_family()[0]
     if family in families:
         commands.append(families[family])
-    elif family in system_fonts and rcParams["pgf.texsystem"] != "pdflatex":
+    elif (any(font.name == family for font in fm.fontManager.ttflist)
+          and rcParams["pgf.texsystem"] != "pdflatex"):
         commands.append(r"\setmainfont{%s}\rmfamily" % family)
     else:
         pass  # print warning?

